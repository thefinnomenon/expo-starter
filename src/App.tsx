import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import DrawerWithTabAndStackNavigator from '@/navigators/DrawerWithTabAndStackNavigator';

export default function App() {
  return (
    <NavigationContainer>
      <DrawerWithTabAndStackNavigator />
    </NavigationContainer>
  );
}
