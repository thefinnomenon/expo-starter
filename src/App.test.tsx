import React from 'react';
import { render } from '@testing-library/react-native';
import App from './App';

jest.mock('aws-amplify');
jest.mock('@/services/sentry');
jest.mock('@/api/amplify');
jest.mock('@/services/location');
jest.mock('@/services/analytics');
jest.mock('@/utilities/testUtility');

describe('<App />', () => {
  it('renders correctly', () => {
    const tree = render(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays process.env.NODE_ENV', () => {
    const view = render(<App />);
    const text = view.getByText(new RegExp('^process.env.NODE_ENV: [a-zA-z0-9]+'));
    expect(text).not.toBeNull();
  });

  it('displays process.env.NAME', () => {
    const view = render(<App />);
    const text = view.getByText(new RegExp('^process.env.NAME: [a-zA-z0-9]+'));
    expect(text).not.toBeNull();
  });
});
