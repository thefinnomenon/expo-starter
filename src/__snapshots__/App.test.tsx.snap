// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    process.env.NODE_ENV: 
    test
  </Text>
  <Text>
    process.env.NAME: 
    test
  </Text>
  <Text>
    Path Alias: 
  </Text>
  <Text>
    - Location -
  </Text>
  <Text>
    Latitude: 
  </Text>
  <Text>
    Longitude: 
  </Text>
  <Text>
    - Reverse Geocode -
  </Text>
  <Text />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press to cause error!
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press to cause analytics event!
      </Text>
    </View>
  </View>
</View>
`;
